### PR TITLE
internal/travis: clean up runchecks.sh

### PR DESCRIPTION
**Pull request by vangent**
Removes the `-x` bash command so that we no longer echo every command, and inserts some explicit `echo`s to keep track of what's going on. IMHO this makes the output a lot more readable.

### CODE REVIEW

File: internal/testing/runchecks.sh [hunk 1]
```diff
@@ -81,53 +90,65 @@ else
...
   result=1
 fi
 
-# Ensure that all examples used in Hugo match what's in source.
+
+echo
+echo "Ensuring that all examples used in Hugo match what's in source..."
 internal/website/gatherexamples/run.sh | diff internal/website/data/examples.json - > /dev/null || {
   echo "FAIL: examples changed; run: internal/website/gatherexamples/run.sh > internal/website/data/examples.json"
+  result=1
```
Comment (vangent) [inline internal/testing/runchecks.sh]: Pretty sure this was a bug before, missing `result=1` meant that this wouldn't actual fail.

**Comment by eliben** [comment]
Adding new echos SGTM, but I'm not sure how removing `-x` makes it more readable. Can you give an example? I recall finding it really useful before to see exactly which commands ran before something failed, etc.

**Comment by vangent** [comment]
> Adding new echos SGTM, but I'm not sure how removing `-x` makes it more readable. Can you give an example? I recall finding it really useful before to see exactly which commands ran before something failed, etc.

Little example:

```
+echo 'Looking for files that changed'
Looking for files that changed
```

Larger example (totally useless text):

```
++gofmt -s -d ./health/health.go ./health/sqlhealth/sqlhealth_test.go ./health/sqlhealth/sqlhealth.go ./health/health_test.go ./samples/server/main.go ./samples/guestbook/inject_gcp.go ./samples/guestbook/inject_azure.go ./samples/guestbook/inject_aws.go ./samples/guestbook/aws/provision_db/main.go ./samples/guestbook/gcp/provision_db/main.go ./samples/guestbook/gcp/deploy/main.go ./samples/guestbook/inject_local.go ./samples/guestbook/localdb/main.go ./samples/guestbook/main.go ./samples/guestbook/wire_gen.go ./samples/gocdk-runtimevar/main.go ./samples/gocdk-blob/main.go ./samples/gocdk-secrets/main.go ./samples/appengine/helloworld.go ./samples/gocdk-pubsub/main.go ./samples/tutorial/main.go ./server/server.go ./server/sdserver/server.go ./server/server_test.go ./server/driver/driver.go ./server/xrayserver/server.go ./doc.go ./aws/aws_test.go ./aws/rds/rds.go ./aws/awscloud/awscloud.go ./aws/awscloud/example_test.go ./aws/aws.go ./postgres/postgres.go ./postgres/rdspostgres/rdspostgres_test.go ./postgres/rdspostgres/rdspostgres.go ./postgres/postgres_test.go ./postgres/cloudpostgres/example_test.go ./postgres/cloudpostgres/cloudpostgres.go ./postgres/cloudpostgres/cloudpostgres_test.go ./runtimevar/runtimevar_test.go ./runtimevar/awsparamstore/awsparamstore.go ./runtimevar/awsparamstore/example_test.go ./runtimevar/awsparamstore/awsparamstore_test.go ./runtimevar/example_test.go ./runtimevar/httpvar/httpvar_test.go ./runtimevar/httpvar/example_test.go ./runtimevar/httpvar/httpvar.go ./runtimevar/gcpruntimeconfig/example_test.go ./runtimevar/gcpruntimeconfig/gcpruntimeconfig.go ./runtimevar/gcpruntimeconfig/gcpruntimeconfig_test.go ./runtimevar/oc_test.go ./runtimevar/example_openvariable_test.go ./runtimevar/constantvar/example_test.go ./runtimevar/constantvar/constantvar_test.go ./runtimevar/constantvar/constantvar.go ./runtimevar/filevar/example_test.go ./runtimevar/filevar/filevar.go ./runtimevar/filevar/filevar_test.go ./runtimevar/etcdvar/etcdvar_test.go ./runtimevar/etcdvar/example_test.go ./runtimevar/etcdvar/etcdvar.go ./runtimevar/drivertest/drivertest.go ./runtimevar/driver/driver.go ./runtimevar/runtimevar.go ./runtimevar/blobvar/blobvar_test.go ./runtimevar/blobvar/example_test.go ./runtimevar/blobvar/blobvar.go ./gcp/gcpcloud/example_test.go ./gcp/gcpcloud/gcpcloud.go ./gcp/gcp_test.go ./gcp/gcp.go ./gcp/cloudsql/cloudsql.go ./blob/s3blob/s3blob_test.go ./blob/s3blob/example_test.go ./blob/s3blob/s3blob.go ./blob/fileblob/fileblob_test.go ./blob/fileblob/fileblob.go ./blob/fileblob/example_test.go ./blob/fileblob/attrs.go ./blob/example_test.go ./blob/oc_test.go ./blob/drivertest/bindata.go ./blob/drivertest/drivertest.go ./blob/example_openbucket_test.go ./blob/blob_test.go ./blob/driver/driver.go ./blob/memblob/example_test.go ./blob/memblob/memblob_test.go ./blob/memblob/memblob.go ./blob/blob.go ./blob/gcsblob/example_test.go ./blob/gcsblob/gcsblob_test.go ./blob/gcsblob/gcsblob.go ./blob/azureblob/example_test.go ./blob/azureblob/azureblob_test.go ./blob/azureblob/azureblob.go ./azure/azuredb/azuredb.go ./azure/azurecloud/azurecloud.go ./tests/aws/app/inject.go ./tests/aws/app/app_test.go ./tests/aws/app/app.go ./tests/aws/app/wire_gen.go ./tests/gcp/app/inject.go ./tests/gcp/app/app_test.go ./tests/gcp/app/app.go ./tests/gcp/app/wire_gen.go ./tests/internal/testutil/testutil.go ./requestlog/stackdriver.go ./requestlog/ncsa.go ./requestlog/requestlog.go ./requestlog/stackdriver_test.go ./requestlog/requestlog_test.go ./requestlog/ncsa_test.go ./pubsub/sub_test.go ./pubsub/pub_test.go ./pubsub/example_test.go ./pubsub/kafkapubsub/example_test.go ./pubsub/kafkapubsub/kafka_test.go ./pubsub/kafkapubsub/kafka.go ./pubsub/awssnssqs/awssnssqs.go ./pubsub/awssnssqs/example_test.go ./pubsub/awssnssqs/awssnssqs_test.go ./pubsub/rabbitpubsub/doc.go ./pubsub/rabbitpubsub/example_test.go ./pubsub/rabbitpubsub/fake_test.go ./pubsub/rabbitpubsub/rabbit.go ./pubsub/rabbitpubsub/rabbit_test.go ./pubsub/rabbitpubsub/amqp.go ./pubsub/benchmark_test.go ./pubsub/drivertest/drivertest.go ./pubsub/pubsub_test.go ./pubsub/mempubsub/conformance_test.go ./pubsub/mempubsub/mem.go ./pubsub/mempubsub/example_test.go ./pubsub/mempubsub/mem_test.go ./pubsub/natspubsub/example_test.go ./pubsub/natspubsub/nats_test.go ./pubsub/natspubsub/nats.go ./pubsub/driver/driver.go ./pubsub/azuresb/azuresb_test.go ./pubsub/azuresb/example_test.go ./pubsub/azuresb/azuresb.go ./pubsub/pubsub.go ./pubsub/acks_test.go ./pubsub/gcppubsub/gcppubsub_test.go ./pubsub/gcppubsub/example_test.go ./pubsub/gcppubsub/gcppubsub.go ./gcerrors/errors_test.go ./gcerrors/errors.go ./internal/contributebot/ghauth.go ./internal/contributebot/sync_test.go ./internal/contributebot/setup.go ./internal/contributebot/process_test.go ./internal/contributebot/sync.go ./internal/contributebot/process.go ./internal/contributebot/main.go ./internal/contributebot/wire_gen.go ./internal/contributebot/webhook/main.go ./internal/batcher/batcher_test.go ./internal/batcher/batcher.go ./internal/cmd/gocdk/flagset_test.go ./internal/cmd/gocdk/launch.go ./internal/cmd/gocdk/add.go ./internal/cmd/gocdk/flagset.go ./internal/cmd/gocdk/main_test.go ./internal/cmd/gocdk/biome_test.go ./internal/cmd/gocdk/build.go ./internal/cmd/gocdk/apply.go ./internal/cmd/gocdk/init.go ./internal/cmd/gocdk/signal_unix.go ./internal/cmd/gocdk/main.go ./internal/cmd/gocdk/serve_test.go ./internal/cmd/gocdk/biome.go ./internal/cmd/gocdk/deploy.go ./internal/cmd/gocdk/signal_windows.go ./internal/cmd/gocdk/build_test.go ./internal/cmd/gocdk/apply_test.go ./internal/cmd/gocdk/serve.go ./internal/gcerr/errorcode_string.go ./internal/gcerr/gcerr_test.go ./internal/gcerr/gcerr.go ./internal/retry/retry.go ./internal/retry/retry_test.go ./internal/escape/escape.go ./internal/escape/escape_test.go ./internal/useragent/useragent.go ./internal/oc/trace_test.go ./internal/oc/metrics.go ./internal/oc/trace.go ./internal/docstore/query.go ./internal/docstore/mongodocstore/mongo.go ./internal/docstore/mongodocstore/query.go ./internal/docstore/mongodocstore/mongo_test.go ./internal/docstore/mongodocstore/codec.go ./internal/docstore/doc.go ./internal/docstore/query_test.go ./internal/docstore/example_test.go ./internal/docstore/dynamodocstore/codec_test.go ./internal/docstore/dynamodocstore/query.go ./internal/docstore/dynamodocstore/query_test.go ./internal/docstore/dynamodocstore/example_test.go ./internal/docstore/dynamodocstore/codec.go ./internal/docstore/dynamodocstore/dynamo.go ./internal/docstore/dynamodocstore/dynamo_test.go ./internal/docstore/firedocstore/codec_test.go ./internal/docstore/firedocstore/query.go ./internal/docstore/firedocstore/doc.go ./internal/docstore/firedocstore/query_test.go ./internal/docstore/firedocstore/codec.go ./internal/docstore/firedocstore/fs_test.go ./internal/docstore/firedocstore/native_codec_test.go ./internal/docstore/firedocstore/fs.go ./internal/docstore/drivertest/drivertest.go ./internal/docstore/docstore_test.go ./internal/docstore/driver/codec_test.go ./internal/docstore/driver/driver.go ./internal/docstore/driver/util.go ./internal/docstore/driver/document.go ./internal/docstore/driver/codec.go ./internal/docstore/driver/util_test.go ./internal/docstore/driver/document_test.go ./internal/docstore/docstore.go ./internal/docstore/memdocstore/codec_test.go ./internal/docstore/memdocstore/query.go ./internal/docstore/memdocstore/mem.go ./internal/docstore/memdocstore/codec.go ./internal/docstore/memdocstore/mem_test.go ./internal/docstore/internal/fields/fields_test.go ./internal/docstore/internal/fields/fold.go ./internal/docstore/internal/fields/fields.go ./internal/docstore/internal/fields/fold_test.go ./internal/openurl/openurl.go ./internal/openurl/openurl_test.go ./internal/trace/trace_test.go ./internal/trace/trace.go ./internal/website/gatherexamples/gatherexamples_test.go ./internal/website/gatherexamples/gatherexamples.go ./internal/testing/octest/exporter.go ./internal/testing/octest/diff.go ./internal/testing/terraform/terraform.go ./internal/testing/setup/setup.go ./internal/testing/replay/replay.go ./mysql/cloudmysql/example_test.go ./mysql/cloudmysql/cloudmysql.go ./mysql/cloudmysql/cloudmysql_test.go ./mysql/mysql.go ./mysql/mysql_test.go ./mysql/rdsmysql/example_test.go ./mysql/rdsmysql/rdsmysql_test.go ./mysql/rdsmysql/rdsmysql.go ./mysql/azuremysql/azuremysql_test.go ./mysql/azuremysql/azuremysql.go ./secrets/secrets_test.go ./secrets/azurekeyvault/example_test.go ./secrets/azurekeyvault/akv.go ./secrets/azurekeyvault/akv_test.go ./secrets/gcpkms/example_test.go ./secrets/gcpkms/kms.go ./secrets/gcpkms/kms_test.go ./secrets/awskms/example_test.go ./secrets/awskms/kms.go ./secrets/awskms/kms_test.go ./secrets/localsecrets/localsecrets.go ./secrets/localsecrets/example_test.go ./secrets/localsecrets/localsecrets_test.go ./secrets/example_test.go ./secrets/vault/vault_test.go ./secrets/vault/example_test.go ./secrets/vault/vault.go ./secrets/drivertest/drivertest.go ./secrets/driver/driver.go ./secrets/secrets.go ./secrets/example_openkeeper_test.go
```

There are also a number of places where we run commands comparing to some variable (e.g., `$DIFF`), which means that the whole variable gets dumped out.

There are situations where you're debugging the actual script where it is useful to see the commands, but I think it's far more frequent to be scanning the file looking for where it failed, and right now it's pretty hard to do that.

**Comment by eliben** [comment]
Adding @zombiezen to TAL too.

I'm not bothered by extra output, but don't mind to try if it really bothers you

**Comment by vangent** [comment]
@eliben waiting for your approval as well.

**Pull request merged**